envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/SnapshotUpdater.kt

```diff
@@ -133,6 +136,20 @@ class SnapshotUpdater(
             logger.error("Unable to create snapshot for group ${group.serviceName}", e)
         }
     }
+
+    private fun updateSnapshotForGroups(groups: Collection<Group>, result: UpdateResult) {
+        versions.retainGroups(cache.groups())
+        groups.forEach { group ->
+            if (result.adsSnapshot != null && group.communicationMode == ADS) {
+                updateSnapshotForGroup(group, result.adsSnapshot)
+            } else if (result.xdsSnapshot != null && group.communicationMode == XDS) {
+                updateSnapshotForGroup(group, result.xdsSnapshot)
+            } else {
+                logger.error("Requested snapshot for ${group.communicationMode.name} mode," +
+                    " but it is not here. This should never happen")
```

Comment:
Should this have a metric and an alarm associated?

Reply:
Added metric